tools/gen_compilation_database.py

```diff
@@ -25,11 +25,21 @@ def generate_compilation_database(args):
     execroot = subprocess.check_output(["bazel", "info", "execution_root"]
                                        + bazel_options).decode().strip()
 
-    compdb = []
-    for compdb_file in Path(execroot).glob("**/*.compile_commands.json"):
-        compdb.extend(
-            json.loads("[" + compdb_file.read_text().replace("__EXEC_ROOT__", execroot) + "]"))
-    return compdb
+    db_entries = []
+    for db in pathlib.Path(execroot).glob('**/*.compile_commands.json'):
+        with open(db, 'r') as f:
+            db_entries.extend(json.load(f))
+
+    def replace_execroot_marker(db_entry):
+        if 'directory' in db_entry and db_entry['directory'] == '__EXEC_ROOT__':
+            db_entry['directory'] = execroot
+        if 'command' in db_entry:
+            db_entry['command'] = (
+                db_entry['command'].replace('-isysroot __BAZEL_XCODE_SDKROOT__', ''))
+        return db_entry
+
+    db_entries = list(map(replace_execroot_marker, db_entries))
```

Comment:
i think just return the `list(map(re...` rather than creating a var

Reply:
Done